src/Microsoft.Framework.ConfigurationModel.Interfaces/IConfigurationSourceRoot.cs

```diff
@@ -7,6 +7,8 @@ namespace Microsoft.Framework.ConfigurationModel
 {
     public interface IConfigurationSourceRoot : IConfiguration
     {
+        string BasePath { get; }
```

Comment:
If the BasePath is in `IConfiguration` it will make it hard to resolve the absolute path when a new source is added, because by that time the `IConfigurationSourceRoot` is given only.